master/static/srv/shell-entrypoint.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+set -e
+
+export PATH="/run/determined/pythonuserbase/bin:$PATH"
+
+# Unlike trial and notebook entrypoints, the HOME directory does not need to be
+# modified in this entrypoint because the HOME in the user's ssh session is set
+# by sshd at a later time.
+
+python3.6 -m pip install --user /opt/determined/wheels/determined*.whl
+
+# Prepend each key in authorized_keys with a set of environment="KEY=VALUE"
```

Comment:
question: Are user defined environment variables at all supported for shells?

Reply:
Whoa, that's a good point.  I'll work on that.

Reply:
After spending some time looking at this, I actually don't think the answer to this is super clear-cut.

1) One thing we could do is just pass the entire environment through sshd through the mechanism listed here, so that the environment inside the shell would be the same as what was specified for the docker container... but I kind of doubt that that is the most correct way to handle environment variables for shells in general, for things like LANG or TERM which should be set based on the environment that the user is passing in (admittedly, there are already good mechanisms for LANG and TERM, specifically).

2) Another thing we could do is not support user-defined variables here at all and have users pass them in via normal shell mechanisms (putting a .bashrc in the context directory I suppose).  But that seems odd since someone with a custom environment would have to pack their environment variables into shells specially.

The pro of 2 over 1 is that 2 it that it's easier to add features later than take them away later.  The pro of 1 over 2 is that it seems to work nicer with the rest of our system, especially in the case of custom images or standardized configuration files for multiple users.

I think I am inclined to do 1.  I did some initial testing to make sure that the mechanisms for option 2 can properly override settings form 1, and I also confirmed that variables like TERM and LANG are overwritten if the client sends them.

What do you think?

Reply:
Given the time crunch, I went ahead and implemented 1 but I'm still open to 2 if you and @dzhu think it is a better idea.

Reply:
Sounds okay to me.